### PR TITLE
FE-1196 - Default Domain Confirmation Modal

### PR DESCRIPTION
**Pull request by nstanard**
### What Changed
 - Added a confirmation modal and cypress tests to the new domain details page.

### How To Test
 - Run the cypress tests for domains/domainDetails.spec.js
 - Load the app the navigate to `http://localhost:3100/domains/details/fad-5415.io` logged in with appteam. Then toggle the "Set as Default Tracking Domain" checkbox.

### To Do
- [ ] Address any feedback/suggestions that the team finds important to this work.

### CODE REVIEW

File: src/pages/domains/components/DomainStatusSection.js [hunk 1]
```diff
@@ -107,19 +112,35 @@ export default function DomainStatusSection({
...
                       Set as Default Tracking Domain <Bookmark color="green" />
                     </>
                   }
                   checked={trackingDomain.defaultTrackingDomain}
-                  onClick={() =>
+                  onClick={() => setConfirmationModalOpen(true)}
+                />
+
+                <ConfirmationModal
+                  open={isConfirmationModalOpen}
```
Comment (shrirupa) [inline src/pages/domains/components/DomainStatusSection.js:126]: We probably should not be using Confirmation Modal here, as it uses the Modal component from pre-hibana matchbox. link to the new matchbox storybook - https://matchbox-storybook.netlify.app/?path=/story/overlays-modal--configurable-buttons. for reference - there is also another Modal in use in VerifyEmailSection.js

File: src/pages/domains/components/DomainStatusSection.js [hunk 2]
```diff
@@ -107,19 +112,35 @@ export default function DomainStatusSection({
...
+                        : `Transmissions and templates that don't specify a tracking domain will now use ${domainName}.`}
+                    </p>
+                  }
+                  isPending={false}
+                  onCancel={() => setConfirmationModalOpen(false)}
+                  onConfirm={() =>
                     toggleDefaultTracking({
-                      domain: trackingDomain?.domainName,
+                      domain: domainName,
                       subaccount: trackingDomain.subaccountId,
```
Comment (shrirupa) [inline src/pages/domains/components/DomainStatusSection.js:140]: my mistake 🙂   but we should probably avoid passing these to the function too, and use trackingDomain?.domainName and trackingDomain.subaccountId directly.

File: src/pages/domains/components/DomainStatusSection.js [hunk 3]
```diff
@@ -19,6 +20,7 @@ export default function DomainStatusSection({
   id,
   isTracking,
 }) {
+  const [isConfirmationModalOpen, setConfirmationModalOpen] = useState(false);
```
Comment (shrirupa) [inline src/pages/domains/components/DomainStatusSection.js:23]: you can ignore this if you want but we have a nice hook called useModal that can be useful here.

File: cypress/tests/integration/domains/detailsPage.spec.js [hunk 4]
```diff
@@ -193,6 +197,124 @@ describe('The domains details page', () => {
...
+      it('confirms removal of default domain tracking.', () => {
+        cy.stubRequest({
+          url: '/api/v1/tracking-domains',
+          fixture: 'tracking-domains/200.get.json',
+          requestAlias: 'trackingDomainsList',
+        });
+
+        cy.visit(`${BASE_UI_URL}/verified-and-default.com`);
+        cy.wait(['@trackingDomainsList']);
+        cy.wait('@accountDomainsReq');
```
Comment (shrirupa) [inline cypress/tests/integration/domains/detailsPage.spec.js:209]: we can group these together - `cy.wait(['@trackingDomainsList', '@accountDomainsReq']);`

File: cypress/tests/integration/domains/detailsPage.spec.js [hunk 5]
```diff
@@ -193,6 +197,124 @@ describe('The domains details page', () => {
...
+        cy.findAllByLabelText('Set as Default Tracking Domain').should('be.checked');
+
+        // todo: why am I having to force click?
+        cy.findAllByLabelText('Set as Default Tracking Domain').click({ force: true });
+
+        cy.withinModal(() => {
+          cy.server();
+
+          cy.route({
+            method: 'PUT',
```
Comment (shrirupa) [inline cypress/tests/integration/domains/detailsPage.spec.js:223]: we can use stubRequest here.

File: cypress/tests/integration/domains/detailsPage.spec.js [hunk 6]
```diff
@@ -193,6 +197,124 @@ describe('The domains details page', () => {
...
+      it('confirms setting a default domain tracking.', () => {
+        cy.stubRequest({
+          url: '/api/v1/tracking-domains',
+          fixture: 'tracking-domains/200.get.json',
+          requestAlias: 'trackingDomainsList',
+        });
+
+        cy.visit(`${BASE_UI_URL}/verified.com`);
+        cy.wait(['@trackingDomainsList']);
+        cy.wait('@accountDomainsReq');
```
Comment (shrirupa) [inline cypress/tests/integration/domains/detailsPage.spec.js:270]: https://github.com/SparkPost/2web2ui/pull/1834/files#r501774666

File: cypress/tests/integration/domains/detailsPage.spec.js [hunk 7]
```diff
@@ -193,6 +197,124 @@ describe('The domains details page', () => {
...
+        cy.findAllByLabelText('Set as Default Tracking Domain').should('not.be.checked');
+
+        cy.findAllByLabelText('Set as Default Tracking Domain').click({ force: true });
+
+        cy.withinModal(() => {
+          cy.server();
+
+          cy.route({
+            method: 'PUT',
+            url: '/api/v1/tracking-domains/verified.com',
```
Comment (shrirupa) [inline cypress/tests/integration/domains/detailsPage.spec.js:284]: https://github.com/SparkPost/2web2ui/pull/1834/files#r501776062

File: src/pages/domains/components/DomainStatusSection.js (hunk 3 above)
Comment (nstanard) [reply src/pages/domains/components/DomainStatusSection.js:23]: I'll check it out, might as well be familiar with it.

File: cypress/tests/integration/domains/detailsPage.spec.js [hunk 8]
```diff
@@ -1,6 +1,9 @@
 import { IS_HIBANA_ENABLED } from 'cypress/constants';
+
 const BASE_UI_URL = '/domains/details';
-const PAGE_URL = '/domains/details/fake-domain.com';
+const PAGE_URL = `${BASE_UI_URL}/fake-domain.com`;
+
+let trackingDomainsList = require('../../../fixtures/tracking-domains/200.get.json');
```
Comment (shrirupa) [inline cypress/tests/integration/domains/detailsPage.spec.js:6]: can we change this to `cypress/fixtures/tracking-domains/200.get.json` ?

Comment (nstanard) [reply cypress/tests/integration/domains/detailsPage.spec.js:6]: Yea, didn't realize we had a folder alias setup for it - thanks!

File: src/pages/domains/components/DomainStatusSection.js [hunk 9]
```diff
@@ -107,19 +113,30 @@ export default function DomainStatusSection({
...
+                  open={isModalOpen}
+                  title={`${isDefault ? 'Remove' : 'Set'} default tracking domain (${domainName})`}
+                  content={
+                    <p>
+                      {isDefault
+                        ? `Transmissions and templates that don't specify a tracking domain will no longer use ${domainName}. Instead, they will use the system default until another default is selected.`
+                        : `Transmissions and templates that don't specify a tracking domain will now use ${domainName}.`}
+                    </p>
                   }
+                  isPending={false}
```
Comment (shrirupa) [inline src/pages/domains/components/DomainStatusSection.js:136]: I am really sorry, I didn't notice this earlier. 

isPending should actually equal to  `deletePending`, and you can add this to Container.js as 
`deletePending: state.trackingDomain.deleting || state.sendingDomain.deleting` in mapStateToProps.
you would have to update this src/reducers/sendingDomains with

>      case 'DELETE_SENDING_DOMAIN_PENDING':
>       return { ...state, deleting: meta.domain, deleteError: null };
> 
>     case 'DELETE_SENDING_DOMAIN_FAIL':
>       return { ...state, deleting: false, deleteError: payload };
> 
>     case 'DELETE_SENDING_DOMAIN_SUCCESS':
> 

you can reference src/reducers/trackingDomain for the same.

Comment (nstanard) [reply src/pages/domains/components/DomainStatusSection.js]: https://github.com/SparkPost/2web2ui/pull/1834/commits/d235c67ba1295b285a8a8c56332ae99d57399972 - Pushed, but the redux stuff and I'm not as confident with.

File: cypress/plugins/index.js [hunk 10]
```diff
@@ -12,16 +12,12 @@ const path = require('path');
...
 const WEBPACK_OPTIONS = {
   mode: 'development',
   resolve: {
     alias: {
-      'src/helpers': path.resolve('.', 'src/helpers'),
-      'src/config': path.resolve('.', 'src/config'),
-      'src/constants': path.resolve('.', 'src/constants'),
-      'cypress/helpers': path.resolve('.', 'cypress/helpers'),
-      'cypress/constants': path.resolve('.', 'cypress/constants'),
+      src: path.join(__dirname, '../../src'),
```
Comment (shrirupa) [inline cypress/plugins/index.js:19]: 🎉 🎉

**Review by shrirupa — APPROVED** [review]
Awesome! 🎉  Looks good to me!!

**Pull request merged**